app/concepts/todo/create.rb

```diff
@@ -0,0 +1,20 @@
+class Todo::Create < Trailblazer::Operation
+	
+	contract do 
+		def prepopulate!(options)
+			@model = options[:current_user] 
+		end
+	end
+
+	def process(params)
+		title = @params[:title] || ''
+		description = @params[:description] || ''
+		if @contract.model
+			@valid = @contract.model.todos.create({title: title, 
+				description: description})
+		else
+			@valid = false
+		end
+			
+	end
```

Comment:
Raymond, almost there! I see that you've done a little bit of reading as your'e already using a little bit of advanced features like `prepopulate`. Here's the simplest way to use trailblazer:

We use contract to manage forms e.g. validate input without touching the real model. We use `validate` inside `process` to perform the validation on `params`. If validation is successful, then we call f.save to save the contents of `params[:thing]` to the `thing` model. Check out this out:

```
class Thing::Create < Trailblazer::Operation
    contract do
        property :name
        property :description

        validates :name, presence: true
        validates :description, length: {min: 20, max: 50}
    end

    def process(params)
        thing = Thing.new
        validate(params[:thing], thing) do |f|
            f.save
        end
    end
end
```

Good going by the way!